hybridCarboniteMetadataService/src/main/scala/cromwell/services/metadata/hybridcarbonite/CarbonitingMetadataFreezerActor.scala

```diff
@@ -0,0 +1,122 @@
+      goto(Fetching) using FetchingData(workflowId)
+  }
+
+  when(Fetching) {
+    case Event(BuiltMetadataResponse(_, responseJson), FetchingData(workflowId)) =>
+      asyncIo.writeAsync(carboniterConfig.makePath(workflowId), responseJson.compactPrint, Seq(StandardOpenOption.CREATE)) onComplete {
+        result => self ! CarbonitingFreezeResult(result)
+      }
+      goto(Freezing) using FreezingData(workflowId)
+
+    case Event(FailedMetadataResponse(_, reason), FetchingData(workflowId)) =>
+      log.error(reason, s"Failed to fetch workflow $workflowId's metadata to archive. Marking as $ArchiveFailed")
+      scheduleDatabaseUpdateAndAwaitResult(workflowId, ArchiveFailed)
+  }
+
+  when(Freezing) {
+    case Event(CarbonitingFreezeResult(freezeResult), FreezingData(workflowId)) =>
+      freezeResult.failed.foreach { reason => log.error(reason, s"Failed to freeze workflow $workflowId's metadata to GCS archive. Marking as $ArchiveFailed") }
+
+      val newStatus = statusFromFreezeResult(freezeResult)
+      scheduleDatabaseUpdateAndAwaitResult(workflowId, newStatus)
+  }
+
+  when(UpdatingDatabase) {
+    case Event(DatabaseUpdateCompleted(Success(_)), UpdatingDatabaseData(_, _)) =>
+      // All set; exit normally:
+      stop(FSM.Normal)
+    case Event(DatabaseUpdateCompleted(Failure(reason)), UpdatingDatabaseData(workflowId, freezingResult)) =>
+      log.error(reason, s"Unable to update workflow ID $workflowId's METADATA_ARCHIVE_STATUS to $freezingResult. Will try again in 1 minute")
+      scheduleDatabaseUpdateAndAwaitResult(workflowId, freezingResult, delay = Option(1.minute))
+  }
+
+  whenUnhandled {
+    case other =>
+      log.error(s"Programmer Error: Unexpected message to ${getClass.getSimpleName} ${self.path.name} in state $stateName with $stateData: $other")
+      stay()
+  }
+
+  def statusFromFreezeResult(freezeResult: Try[Unit]): MetadataArchiveStatus = freezeResult match {
+    case Success(_) => Archived
+    case Failure(_) => ArchiveFailed
+  }
+
+  def scheduleDatabaseUpdateAndAwaitResult(workflowId: WorkflowId, newStatus: MetadataArchiveStatus, delay: Option[FiniteDuration] = None) = {
+
+    def updateDatabase() = {
+      val dbUpdateFuture = updateMetadataArchiveStatus(workflowId, newStatus)
+      dbUpdateFuture onComplete { dbUpdateResult => self ! DatabaseUpdateCompleted(dbUpdateResult) }
+    }
+    delay match {
+      case Some(d) => context.system.scheduler.scheduleOnce(d)(updateDatabase())
+      case None => updateDatabase()
+    }
+    goto(UpdatingDatabase) using UpdatingDatabaseData(workflowId, newStatus)
+  }
+}
+
+
+object CarbonitingMetadataFreezerActor {
+  sealed trait CarbonitingMetadataFreezingState
+  case object Pending extends CarbonitingMetadataFreezingState
+  case object Fetching extends CarbonitingMetadataFreezingState
+  case object Freezing extends CarbonitingMetadataFreezingState
+  case object UpdatingDatabase extends CarbonitingMetadataFreezingState
```

Comment:
[TOL] Do you envision the metadata deletion being called in `UpdatingDatabase` or in a new deleting state to be added later?

I see that we need to update the database with success and failure alike, so obviously we wouldn't always delete in `scheduleDatabaseUpdateAndAwaitResult`